rosbag2_test_common/include/rosbag2_test_common/publication_manager.hpp

```diff
@@ -71,31 +72,89 @@ class PublicationManager
             std::this_thread::sleep_for(interval);
           }
         }
-      });
+      };
+
+    publishers_.push_back(std::make_pair(publisher, publisher_fcn));
   }
 
   void run_publishers(bool verbose = false) const
   {
-    for (const auto & pub_fcn : publishers_) {
-      pub_fcn(verbose);
+    for (const auto & pub : publishers_) {
+      std::get<PublicationF>(pub)(verbose);
     }
   }
 
   void run_publishers_async(bool verbose = false) const
   {
     std::vector<std::future<void>> futures;
-    for (const auto & pub_fcn : publishers_) {
-      futures.push_back(std::async(std::launch::async, pub_fcn, verbose));
+    for (const auto & pub : publishers_) {
+      futures.push_back(std::async(std::launch::async, std::get<PublicationF>(pub), verbose));
     }
     for (auto & publisher_future : futures) {
       publisher_future.get();
     }
   }
 
+  /// \brief Wait until publisher with specified topic will be connected to the subscribers or
+  ///    timeout occur.
+  /// \tparam Timeout Data type for timeout duration from std::chrono:: namespace
+  /// \param topic_name topic name to find corresponding publisher
+  /// \param timeout Maximum time duration during which discovery should happen.
+  /// \param n_subscribers_to_match Number of subscribers publisher should have for match.
+  /// \return true if not find publisher by topic name or publisher has specified number of
+  ///    subscribers, otherwise false.
```

Comment:
Ups. I overlooked mistake in description for return value. It should be other way around. i.e. return false if fail and true if success. 
@Barry-Xu-2018 Could you please fix it before merge?

Reply:
Thank you.  
The description about return value is incorrect.